File: src/types/task.ts
import type { Segment } from './segment';

export interface Task {
  id: string;
  project_id: string;
  project_name?: string;
  audio_url: string;
  video_url?: string;
  assignee_id: string;
  verifier_id?: string;
  status: 'pending' | 'in_progress' | 'verification' | 'rework' | 'done' | 'exported';
  priority: 'low' | 'medium' | 'high';
  deadline?: string;
  duration?: number;
  created_at: string;
  updated_at: string;
  segments: Segment[];
}

File: backend/internal/models/task.go
package models

import "time"

type Task struct {
	ID         int64     `json:"id"`
	ProjectID  int64     `json:"project_id"`
	AudioURL   string    `json:"audio_url"`
	AssigneeID *int64    `json:"assignee_id"`
	VerifierID *int64    `json:"verifier_id"`
	Status     string    `json:"status"` 
	CreatedAt  time.Time `json:"created_at"`
}